### PR TITLE
webgpu: Use outShapeStrides in more places

### PR DESCRIPTION
**Pull request by qjia7**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4967)

### CODE REVIEW

**Comment by qjia7** [comment]
@axinging @xhcao @haoyunfeix @gyagp, PTAL

File: tfjs-backend-webgpu/src/shader_preprocessor.ts [hunk 1]
```diff
@@ -233,28 +233,55 @@ function getSetOutputSnippet(
   }
 
   if (outRank >= 2) {
+    switch (outRank) {
+      case 2:
+        snippet += `
+        int getOutputFlatIndex(ivec2 coords) {
+          return int(dot(coords, ivec2(outShapeStrides, 1.)));
```
Comment (axinging) [inline tfjs-backend-webgpu/src/shader_preprocessor.ts:240]: LTGM.
nit, the "1." can be "1"

Comment (qjia7) [reply tfjs-backend-webgpu/src/shader_preprocessor.ts]: Done.

**Comment by qjia7** [comment]
@lina128 @jinjingforever @kainino0x Please take a look, thanks.

**Review by kainino0x — APPROVED** [review]

**Pull request merged**